cameraview/src/main/java/com/otaliastudios/cameraview/Camera1.java

```diff
@@ -684,6 +684,9 @@ private void endVideoImmediately() {
             mCameraCallbacks.dispatchOnVideoTaken(mVideoFile);
             mVideoFile = null;
         }
+        if (mCamera != null) {
+            mCamera.setPreviewCallbackWithBuffer(this);
```

Comment:
Could you move this a few lines above, in the `(mVideoFile != null)` block, after dispatching? We don't want this extra allocation in other cases.

Could you also add mCamera.setPreviewCallbackWithBuffer(null) before? I'm not sure it is needed but that's how we do it in other parts of this class.